### PR TITLE
[2017.7] Merge forward from 2016.11 to 2017.7

### PR DESCRIPTION
**Pull request by rallytime**
Conflicts:
  - python/tornado.sls

### CODE REVIEW

File: python/tornado.sls [hunk 1]
```diff
@@ -3,12 +3,11 @@ include:
   - python.pip
 {%- endif %}
 
-{%- set pinned_pkg = 'tornado>=4.2.1,<4.5.0' %}
+{%- set pinned_pkg = 'tornado>=4.4.3,<4.5.0' %}
```
Comment (gtmanfred) [inline python/tornado.sls]: can we revert this change?

or just remove pinning the package all together and see if it fixes the centos 7 py2 segfault?

Comment (rallytime) [reply python/tornado.sls]: Yep. Maybe we should just stick with the way 2016.11 had it. You said the bug for 4.5.0 is fixed in tornado, right?

Comment (gtmanfred) [reply python/tornado.sls]: yeah, that should be fixed.
